containers/python/pyspark_container.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import print_function
+import rpc
+import os
+import sys
+import json
+
+import numpy as np
+np.set_printoptions(threshold=np.nan)
+
+sys.path.append(os.path.abspath("/lib/"))
+from clipper_admin import cloudpickle
+
+import findspark
+findspark.init()
+import pyspark
+from pyspark import SparkConf, SparkContext
+from pyspark.sql import SparkSession
+import importlib
+
+IMPORT_ERROR_RETURN_CODE = 3
+
+
+def load_predict_func(file_path):
+    with open(file_path, 'r') as serialized_func_file:
+        return cloudpickle.load(serialized_func_file)
+
+
+def load_pyspark_model(metadata_path, spark, model_path):
+    with open(metadata_path, "r") as metadata:
+        metadata = json.load(metadata)
+        if "model_class" not in metadata:
+            print("Malformed metadata file.")
+            sys.exit(1)
+        model_class = metadata["model_class"]
+
+        print("Loading %s model from %s" % (model_class, model_path))
+        splits = model_class.split(".")
+        module = ".".join(splits[:-1])
+        class_name = splits[-1]
+        ModelClass = getattr(importlib.import_module(module), class_name)
+        if model_class == "pyspark.ml.pipeline.PipelineModel":
+            model = ModelClass.load(model_path)
+        else:
+            model = ModelClass.load(spark.sparkContext, model_path)
+    return model
+
+
+class PySparkContainer(rpc.ModelContainerBase):
+    def __init__(self, path, input_type):
+        self.input_type = rpc.string_to_input_type(input_type)
+        predict_fname = "predict_func.pkl"
+        predict_path = "{dir}/{predict_fname}".format(
+            dir=path, predict_fname=predict_fname)
+        self.predict_func = load_predict_func(predict_path)
+        self.spark = SparkSession\
+            .builder\
+            .appName("clipper-pyspark")\
+            .getOrCreate()
+        metadata_path = os.path.join(path, "metadata.json")
+        spark_model_path = os.path.join(path, "pyspark_model_data")
+        self.model = load_pyspark_model(metadata_path, self.spark,
+                                        spark_model_path)
+
+    def predict_ints(self, inputs):
+        if self.input_type != rpc.INPUT_TYPE_INTS:
```

Comment:
Oh I copied this code from the `python_container.py` code. I didn't realize it was redundant. I'll remove it.